GenericTrigger.lua

```diff
@@ -267,22 +265,71 @@ function ConstructFunction(prototype, trigger, inverse)
 end
 
 function WeakAuras.EndEvent(id, triggernum, force)
-  local data = events[id] and events[id][triggernum];
-  if(data) then
-    data.region:DisableTrigger(triggernum);
-    if(timers[id] and timers[id][triggernum]) then
-      timer:CancelTimer(timers[id][triggernum].handle, true);
-      timers[id][triggernum] = nil;
-    end
-  end
-  if not(force) then
-    WeakAuras.SetEventDynamics(id, triggernum, data, true);
-  end
+  local allStates = WeakAuras.GetTriggerStateForTrigger(id, triggernum);
+  allStates[""] = allStates[""] or {};
+  local state = allStates[""];
+
+  state.show = false;
+  state.changed = true;
 end
 
 function WeakAuras.ActivateEvent(id, triggernum, data)
-  WeakAuras.SetEventDynamics(id, triggernum, data);
-  data.region:EnableTrigger(triggernum);
```

Comment:
Same for the generic trigger, instead of directly changing the region, the triggerState is changed and then UpdatedTriggerState is called.